core/domain/suggestion_domain.py

```diff
@@ -72,3 +66,196 @@ def to_dict(self):
             'change_cmd': self.change_cmd,
             'score_category': self.score_category
         }
+
+    @classmethod
+    def from_dict(cls):
+        """Return a Suggestion object of type from a dict."""
+        raise NotImplementedError(
+            'Subclasses of BaseSuggestion should implement from_dict.')
+
+    def validate(self):
+        """Validates the BaseSuggestion object. Each subclass must implement
+        this function
+
+        Raises:
+            ValidationError: One or more attributes of the BaseSuggestion object
+                are invalid.
+        """
+        if (
+                self.suggestion_type not in
+                suggestion_models.SUGGESTION_TYPE_CHOICES):
+            raise utils.ValidationError(
+                'Expected suggestion_type to be among allowed choices, '
+                'recieved %s' % self.suggestion_type)
```

Comment:
Nit: "received" is misspelled. Here and below.

Reply:
Fixed in all places.